src/lib/netlink/bond.rs

```
@@ -522,44 +488,35 @@ pub(crate) fn parse_bond_subordinate_info(raw: &[u8]) -> BondSubordinateInfo {
     let mut ad_aggregator_id = None;
     let mut ad_actor_oper_port_state = None;
     let mut ad_partner_oper_port_state = None;
-
-    while i < raw.len() {
-        let hdr_ptr = raw.as_ptr().wrapping_offset(i.try_into().unwrap());
-        let hdr = parse_nla_header(hdr_ptr);
-        let data_ptr = raw
-            .as_ptr()
-            .wrapping_offset((i + NL_ATTR_HDR_LEN).try_into().unwrap());
-        let data = unsafe {
-            slice::from_raw_parts(data_ptr, hdr.nla_len - NL_ATTR_HDR_LEN)
-        };
-        match hdr.nla_type {
-            IFLA_BOND_SLAVE_STATE => {
-                subordinate_state = parse_as_u8(data).into()
-            }
-            IFLA_BOND_SLAVE_MII_STATUS => mii_status = parse_as_u8(data).into(),
-            IFLA_BOND_SLAVE_LINK_FAILURE_COUNT => {
-                link_failure_count = parse_as_u32(data)
-            }
-            IFLA_BOND_SLAVE_PERM_HWADDR => {
-                perm_hwaddr = parse_as_mac(hdr.data_len, data);
-            }
-            IFLA_BOND_SLAVE_QUEUE_ID => queue_id = parse_as_u16(data),
-            IFLA_BOND_SLAVE_AD_AGGREGATOR_ID => {
-                ad_aggregator_id = Some(parse_as_u16(data));
-            }
-            IFLA_BOND_SLAVE_AD_ACTOR_OPER_PORT_STATE => {
-                ad_actor_oper_port_state = Some(parse_as_u8(data));
-            }
-            IFLA_BOND_SLAVE_AD_PARTNER_OPER_PORT_STATE => {
-                ad_partner_oper_port_state = Some(parse_as_u16(data));
-            }
-            _ => {
-                eprintln!("unknown nla_type {} data: {:?}", hdr.nla_type, data);
+    for nla in nlas {
+        if let Ok(nla) = nla {
```

Please use eprintln!() for errors instead of silently ignore them.